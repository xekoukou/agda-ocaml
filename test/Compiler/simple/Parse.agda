module Parse where

open import Common.Unit
open import Common.Char    
open import Common.String 
open import Common.List  


open import Common.IO            

parse : List String → List Char → String
parse (e ∷ [])     []               = "ha"
parse (e ∷ [])     (')' ∷ xs)       = "ho"
parse (e ∷ es)     (a   ∷ xs)       = parse (e ∷ es) xs
parse _ _ = "hi"


parseRegExp : String
parseRegExp = parse ("ff" ∷ []) ('a' ∷ [])

main : _
main = do
    let w = parseRegExp
    putStrLn w
