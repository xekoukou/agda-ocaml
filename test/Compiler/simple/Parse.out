EXECUTED_PROGRAM

ret > ExitSuccess
out > ha
out >
